failing cache test on MacOS

While locally testing wheels from #323, I encountered a single test failure related to caching. It is likely a pre-existing issue since we do not seem to have been testing on mac previously (see #324)

The failure is in `test_pyfftw_interfaces_cache.py::CacheTest::test_objects_removed_after_keepalive`:

<details>

```
=============================================================== FAILURES ================================================================
____________________________________________ CacheTest.test_objects_removed_after_keepalive _____________________________________________

self = <tests.test_pyfftw_interfaces_cache.CacheTest testMethod=test_objects_removed_after_keepalive>

    @unittest.skipIf(*miss('64'))
    def test_objects_removed_after_keepalive(self):
        _cache = interfaces.cache._Cache()
    
        key = 'the key'
    
        test_array = numpy.random.randn(16)
        obj = builders.fft(test_array)
        _cache.insert(obj, key)
    
        self.assertIs(_cache.lookup(key), obj)
    
        keepalive_time = _cache.keepalive_time
    
        if os.name == 'nt':
            # A hack to keep appveyor from falling over here. I suspect the
            # contention is too much to work properly. Either way, let's
            # assume it's a windows problem for now...
            time.sleep(keepalive_time * 8)
        else:
            # Relax a bit more otherwise
            time.sleep(keepalive_time * 4)
    
        self.assertRaises(KeyError, _cache.lookup, key)
    
        _cache.insert(obj, key)
        old_keepalive_time = _cache.keepalive_time
        _cache.set_keepalive_time(old_keepalive_time * 4)
    
        self.assertIs(_cache.lookup(key), obj)
    
        time.sleep(old_keepalive_time * 3.5)
        # still should be there
        self.assertIs(_cache.lookup(key), obj)
    
        if os.name == 'nt':
            # As above, but with a bit longer
            time.sleep(old_keepalive_time * 16)
        else:
            time.sleep(old_keepalive_time * 8)
    
>       self.assertRaises(KeyError, _cache.lookup, key)
E       AssertionError: KeyError not raised by lookup

test_pyfftw_interfaces_cache.py:328: AssertionError
=========================================================== warnings summary ============================================================
test_r2r.py:125
  /Users/glee/Downloads/pyFFTW_wheels/pyFFTW/tests/test_r2r.py:125: PytestCollectionWarning: cannot collect test class 'TestRealTransform' because it has a __init__ constructor (from: tests/test_r2r.py)
    class TestRealTransform(object):

tests/test_pyfftw_dask_interface.py: 408 warnings
  /Users/glee/mambaforge/envs/py39/lib/python3.9/site-packages/dask/array/core.py:1503: FutureWarning: The `numpy.alltrue` function is not implemented by Dask array. You may want to use the da.map_blocks function or something similar to silence this warning. Your code may stop working in a future release.
    warnings.warn(

tests/test_pyfftw_dask_interface.py: 120 warnings
  /Users/glee/mambaforge/envs/py39/lib/python3.9/site-packages/pyfftw/builders/_utils.py:204: ComplexWarning: Casting complex values to real discards the imaginary part
    a = numpy.asarray(a, dtype=_rc_dtype_pairs[a.dtype.char])

tests/test_pyfftw_dask_interface.py: 76 warnings
  /Users/glee/mambaforge/envs/py39/lib/python3.9/site-packages/numpy/fft/_pocketfft.py:70: ComplexWarning: Casting complex values to real discards the imaginary part
    r = pfi.execute(a, is_real, is_forward, fct)

tests/test_pyfftw_dask_interface.py: 44 warnings
  /Users/glee/mambaforge/envs/py39/lib/python3.9/site-packages/numpy/fft/_pocketfft.py:73: ComplexWarning: Casting complex values to real discards the imaginary part
    r = pfi.execute(a, is_real, is_forward, fct)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================================================== short test summary info ========================================================
FAILED test_pyfftw_interfaces_cache.py::CacheTest::test_objects_removed_after_keepalive - AssertionError: KeyError not raised by lookup
================================== 1 failed, 1789 passed, 9 skipped, 649 warnings in 557.76s (0:09:17) ==================================
```

<\details>

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -19,9 +19,6 @@ test-command = "pytest --import-mode=append {project}"
 
     [tool.cibuildwheel.macos]
 
-    # See: https://github.com/pyFFTW/pyFFTW/issues/327
-    test-skip = "*"
-
     archs = [
         "x86_64",
 

--- a/tests/test_pyfftw_interfaces_cache.py
+++ b/tests/test_pyfftw_interfaces_cache.py
@@ -319,11 +319,7 @@ class CacheTest(unittest.TestCase):
         # still should be there
         self.assertIs(_cache.lookup(key), obj)
 
-        if os.name == 'nt':
-            # As above, but with a bit longer
-            time.sleep(old_keepalive_time * 16)
-        else:
-            time.sleep(old_keepalive_time * 8)
+        time.sleep(old_keepalive_time * 16)
 
         self.assertRaises(KeyError, _cache.lookup, key)
 

